### PR TITLE
Issue 7 Done! Currency Converter built via CMake.

### PR DESCRIPTION
**Pull request by PicklesRcool**
- Added CMakeLists.txt
- Added app witch fetches currency api from web via cURL
- App features command line arguments input
- If no arguments are provided, a help screen is displayed

### CODE REVIEW

File: src/exchange-rates.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include "../include/exchange-rates.h"
+
+ExchangeRates::ExchangeRates() {
+    _base = "EUR";
```
[the remark this answers is not in the document]
Comment (iramykytyn) [reply src/exchange-rates.cpp]: "EUR" can't be just a string here. It should be define

**Comment by iramykytyn** [comment]
Some other things to think about.
What happens if the user enters wrong data? for example string instead of number? what is the user doesn't enter all parameters? There should be validation for such cases.

**Comment by PicklesRcool** [comment]
Test comment just to see what happens...

File: include/converter-app.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#include <vector>
+#include <string>
+#include <iostream>
+
```
[the remark this answers is not in the document]
Comment (PicklesRcool) [reply include/converter-app.h]: Removed

File: main.cpp [hunk 3]
```diff
@@ -1,8 +1,18 @@
+//GET https://api.exchangeratesapi.io/latest HTTP/1.1
```
[the remark this answers is not in the document]
Comment (PicklesRcool) [reply main.cpp]: It was for easier writing function for the GET request to not switch tabs. 
Removed now

File: main.cpp [hunk 4]
```diff
@@ -1,8 +1,18 @@
...
-	std::cout << "Hello World!\n";
 
-	return 0;
-}
+#include "include/converter-app.h"
+
 
+int main(int argc, char* argv[])
+{
+    std::cout << '\n';
```
[the remark this answers is not in the document]
Comment (PicklesRcool) [reply main.cpp]: I wanted a newline before app execution. But its not needed.
Removed.

File: include/converter-app.h [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+#include <curl/curl.h>
+
+#include "exchange-rates-parser.h"
+#include "currency-converter.h"
+
+class ConverterApp {
+public:
+    ConverterApp() = delete;
```
[the remark this answers is not in the document]
Comment (PicklesRcool) [reply include/converter-app.h]: The creation of `ConverterApp` object makes sense only when it has input arguments. However, I guess it's more logical to make the default constructor give the object an empty argument list.
Changed.

File: src/converter-app.cpp [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    double amount;
+
+    if (options[0][0] == '-') {
+        int equalIndex = options[0].find("=", 0, 1);
+
+        curr = options[0].substr(1, equalIndex - 1);
+
+        amount = std::atof(options[0].substr(equalIndex + 1, options[0].size() - equalIndex - 1).c_str());
+    }
+    for (char& c: curr) c = std::toupper(c); // just make uppercase, like in the API
```
[the remark this answers is not in the document]
Comment (PicklesRcool) [reply src/converter-app.cpp]: Fixed. Changed the structure completely. Now it iterates between each argument and puts it into a `parseOption()` or  `parseArgument` method.

File: src/currency-converter.cpp [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+std::map<std::__cxx11::string, double> CurrencyConverter::rates() const
+{
+    return _rates._rates;
+}
+
+double CurrencyConverter::convert(const std::string fromCurr, const double amount, const std::string toCurr)
+{
+    double coff = _rates._rates[toCurr] / _rates._rates[fromCurr];
```
[the remark this answers is not in the document]
Comment (PicklesRcool) [reply src/currency-converter.cpp]: Added validation

File: src/exchange-rates-parser.cpp [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    resBase = text.substr(baseIndex, 3);
+
+    resRates[resBase] = 1.f;
+
+    startIndex = text.find("rates", 0, 5) + 9; // 9 in the distance to the first currency
+
+    std::string nextCurr;
+    double nextValue = 0;
+
+    for (int nextIndex = 0; text[nextIndex] != '}';) {
```
[the remark this answers is not in the document]
Comment (PicklesRcool) [reply src/exchange-rates-parser.cpp]: Added parsing macros.

File: src/exchange-rates.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (PicklesRcool) [reply src/exchange-rates.cpp]: Added `DEFAULT_CURRENCY` macros and set it to USD

**Pull request merged**